pkgs/by-name/gd/gdmd/package.nix

```
@@ -0,0 +1,50 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+, makeWrapper
+, gdc
+, perl
+}:
+stdenvNoCC.mkDerivation {
+  pname = "gdmd";
+  version = "0.1.0-unstable-2024-05-30";
+
+  src = fetchFromGitHub {
+    owner = "D-Programming-GDC";
+    repo = "gdmd";
+    rev = "dc0ad9f739795f3ce5c69825efcd5d1d586bb013";
+    hash = "sha256-Sw8ExEPDvGqGKcM9VKnOI6MGgXW0tAu51A90Wi4qrRE=";
+  };
+
+  patches = [
+    ./0001-gdc-store-path.diff
+  ];
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ gdc perl ];
+
+  postPatch = ''
+    substituteInPlace dmd-script \
+      --replace-fail gdcStorePath ${gdc}
```

This is not required, but I think it's best practice:

When creating patches, variables that are supposed to be populated are typically surrounded by `@` symbols (usually not to replace something unrelated accidentally).
So the patch file would have something like `@gcdStorePath@`.

The `substitute*` commands have the `--subst-var` and `--subst-var-by` flags that do this replacement.

```suggestion
      --subst-var-by gdcStorePath ${gdc}
```

or alternatively you could update the patch file to take in some `@gdc_dir@` directly

```suggestion
      --subst-var-by gdc_dir ${gdc}/bin
```

---

Ah I never knew about that, thanks.